src/path_provider.cpp

```diff
@@ -0,0 +1,153 @@
+
+#include <core/project_settings.h>
+#include <core/os/dir_access.h>
+#include "path_provider.h"
+#include "gd_kotlin.h"
+
+String PathProvider::provide_runtime_usercode_path() {
+#ifdef TOOLS_ENABLED
+    String build_lock_dir_path = provide_build_lock_dir_path();
+#if defined(__linux__) || defined(__APPLE__)
+    String usercode_jar_path{build_lock_dir_path + "/" + get_usercode_name()};
+#elif defined _WIN32 || defined _WIN64
+    String usercode_jar_path{build_lock_dir_path + "\\" + get_usercode_name()};
+#endif
+    return usercode_jar_path;
+#else //TOOLS_ENABLED
+    return ProjectSettings::get_singleton()->globalize_path("user://" + get_usercode_name());
+#endif //TOOLS_ENABLED
+}
+
+bool PathProvider::copy_usercode_jar_if_necessary() {
```

Comment:
really weird to have this function in PathProvider.  I would put it back in GDKotlin for now.